app/src/main/java/org/android/go/sopt/presentation/auth/SignUpActivity.kt

```diff
@@ -1,96 +1,90 @@
+        binding.btnSignupComplete.isEnabled = viewModel.checkSignUpValidation()
     }
 
+    private fun observe() {
+        viewModel.signUpResult.observe(this) {
+            startActivity(Intent(this, SignInActivity::class.java))
+            finish()
+        }
 
-    private fun completeSignUp() {
-        binding.btnSignupComplete.setOnClickListener {
-            signUpService.signUp(
-                with(binding) {
-                    RequestSignUpDto(
-                        etSignupId.text.toString(),
-                        etSignupPw.text.toString(),
-                        etSignupName.text.toString(),
-                        etSignupSpecialty.text.toString()
-                    )
-                }
-            ).enqueue(object : retrofit2.Callback<ResponseSignUpDto> {
-                override fun onResponse(
-                    call: Call<ResponseSignUpDto>,
-                    response: Response<ResponseSignUpDto>,
-                ) {
-                    if (response.isSuccessful && canUserSignUp()) {
-                        response.body()?.message?.let { binding.root.showToast(it) }
-                            ?: "회원가입에 성공했습니다."
-                        finish()
-
-                        if (!isFinishing) finish()
-                    } else if (!canUserSignUp()) {
-                        binding.root.showToast("회원가입에 실패했습니다.")
-                    } else {
-                        // 응답 실패!
-                        response.body()?.message?.let { binding.root.showToast(it) }
-                            ?: "서버통신 실패(40X)"
-                    }
+        viewModel.id.observe(this) {
+            with(binding) {
+                if (viewModel!!.checkIdFormation()) {
+                    tvSignUpIdWaring.visibility = View.INVISIBLE
+                    etSignupId.setBackgroundResource(R.drawable.rectangle_grey_500_radius_8)
+                } else {
+                    tvSignUpIdWaring.visibility = View.VISIBLE
+                    etSignupId.setBackgroundResource(R.drawable.rectangle_red_500_radius_8)
                 }
+            }
+        }
 
-                override fun onFailure(call: Call<ResponseSignUpDto>, t: Throwable) {
-                    // 왜 안 오지?!
-                    t.message?.let { binding.root.showToast(it) } ?: "서버통신 실패(응답값 X)"
+        viewModel.pw.observe(this) {
+            with(binding) {
+                if (viewModel!!.checkPwFormation()) {
+                    tvSignUpPwWaring.visibility = View.INVISIBLE
+                    etSignupPw.setBackgroundResource(R.drawable.rectangle_grey_500_radius_8)
+                } else {
+                    tvSignUpPwWaring.visibility = View.VISIBLE
+                    etSignupPw.setBackgroundResource(R.drawable.rectangle_red_500_radius_8)
                 }
-            })
-
+            }
         }
 
-
+        viewModel.isEnabledSignUpButton.observe(this) {
+            binding.btnSignupComplete.isEnabled = it
+        }
```

Comment:
```suggestion
        viewModel.isEnabledSignUpButton.observe(this) { isEnable ->
            binding.btnSignupComplete.isEnabled = isEnable
        }
```
스코프 함수에서 `it` 대신 명시적인 네이밍을 써주면 조금 더 가독성 좋은 코드를 쓸 수 있어요!

Reply:
오!!! 좋은 피드백 감사합니다ㅎㅎ